torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -212,6 +212,19 @@ std::pair<at::Tensor, std::shared_ptr<ir::Value>> ReduceScatter(
       std::make_shared<ir::Value>(new_token));
 }
 
+std::shared_ptr<ir::Value> ReduceScatterOut(
+    const std::string& reduce_type, at::Tensor& output, const at::Tensor& input,
+    const std::shared_ptr<ir::Value>& token, double scale, int64_t scatter_dim,
+    int64_t shard_count,
+    const std::vector<std::vector<int64_t>>& replica_groups) {
+  XLATensor out = bridge::GetXlaTensor(output);
```

Comment:
I came here for inspiration of how to set the `ir::Value` for an output tensor. Here I agree with Milad: I don't believe we need to leave the `out` variable outside of the `reduce_scatter_out` call. Leaving `out` outside doesn't do anything -- `out` will be destructed immediately after the return statement anyway.

`XLATensor` is a holder for an `shared_ptr<Data>`, so it can be used as a temp. `XLATensor::SetIrValue` (and any other non-const method on `XLATensor`) manipulates the `shared_ptr<Data>` member inside `XLATensor`, so passing `bridge::GetXlaTensor(output)` is fine.

Reply:
NVM, I found the compiler doesn't like it :D